spinn_front_end_common/interface/interface_functions/router_provenance_gatherer.py

```diff
@@ -88,23 +129,28 @@ def __call__(
         self._add_router_provenance_data(
             router_tables, extra_monitor_vertices, prov_items)
 
-        prov_items.append(ProvenanceDataItem(
+        prov_items.extend(self.__summary_items())
+        return prov_items
+
+    def __summary_items(self):
+        """
+        :rtype: iterable(ProvenanceDataItem)
+        """
+        yield ProvenanceDataItem(
             ["router_provenance", "total_multi_cast_sent_packets"],
-            self._total_sent_packets))
-        prov_items.append(ProvenanceDataItem(
+            self._total_sent_packets)
+        yield ProvenanceDataItem(
```

Comment:
using yield instead of building lists to only use in an extend call is APPROVED!